packages/docusaurus-plugin-content-docs/src/sidebars/index.ts

```diff
@@ -91,5 +98,28 @@ export async function loadSidebars(
     sidebarFilePath,
     normalizeSidebarsParams,
   );
-  return processSidebars(normalizedSidebars, options);
+  const categoriesMetadata = Object.fromEntries(
+    await Promise.all(
+      (
+        await Globby('**/_category_.{json,yml,yaml}', {
```

Comment:
why not keep the former code instead?

If 2 files exist with the same extension, which extension "wins"? 

Wouldn't we load/validate both objects when we could only load/validate one?

Does the order of categoriesMetadata attributes matter now? 

Is this order defined implicitly by the underlying Globby lib and could change from one day to another? 

Is it reliable per FS/OS?

Reply:
The resolution order doesn't matter, it's undefined behavior. We never guaranteed which one will be picked.

Reply:
Yes it was definitively explicitly defined before:

```js
const filePath = await findAsyncSequential(
    ['.json', '.yml', '.yaml'].map((ext) =>
      posixPath(
        path.join(categoryDirPath, `${CategoryMetadataFilenameBase}${ext}`),
      ),
    ),
    fs.pathExists,
  );
```

And it does really matter: we don't want non-deterministic behavior. 

I don't think it'll happen for this use-case, but we don't want to lose 1 day later this year on debugging a user issue that is related to duplicate config file and different envs (ie it works on dev computer, but not on CI for example)

Reply:
Oh, I don't like these speculations😄

- The `find` was sequential, only because parallel find is too intractable and not worth the complication. If I could, I would make it parallel.
- Would there ever be a user who has both `_category_.json` and `_category_.yml` and expect it to always work? Would it take more than one minute to debug?
- "Fixing" a bug by literally "crystallizing" it and saying "this is very, very bad practice but you can go on, we got your back" is much worse than "you played with fire, now you are burnt". I'd rather throw/warn than have some "deterministic behavior" for bad configuration in the first place.
- Undefined behavior is an important concept in any system. UB draws the boundaries and allows us to make important optimizations. It's the same as keeping things undocumented: anything undocumented is not encouraged, and we can break your code without warning.

For this particular case, I would just warn the user about UB, but we have many UBs with clashes: clashing routes, for example. These are important features of a system and not always something we need to fix.